src/EditorFeatures/CSharpTest/SplitComment/SplitCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,268 @@
+
+                if (verifyUndo)
+                {
+                    // Ensure that after undo we go back to where we were to begin with.
+                    var history = undoHistoryRegistry.GetHistory(document.GetTextBuffer());
+                    history.Undo(count: originalSelections.Count);
+
+                    var currentSnapshot = document.GetTextBuffer().CurrentSnapshot;
+                    Assert.Equal(originalSnapshot.GetText(), currentSnapshot.GetText());
+                }
+            }
+        }
+
+        /// <summary>
+        /// verifyUndo is needed because of https://github.com/dotnet/roslyn/issues/28033
+        /// Most tests will continue to verifyUndo, but select tests will skip it due to
+        /// this known test infrastructure issure. This bug does not represent a product
+        /// failure.
+        /// </summary>
+        private void TestHandled(
+            string inputMarkup, string expectedOutputMarkup,
+            bool verifyUndo = true, IndentStyle indentStyle = IndentStyle.Smart)
+        {
+            TestWorker(
+                inputMarkup, expectedOutputMarkup,
+                callback: () =>
+                {
+                    Assert.True(false, "Should not reach here.");
+                },
+                verifyUndo, indentStyle);
+        }
+
+        private void TestNotHandled(string inputMarkup)
+        {
+            var notHandled = false;
+            TestWorker(
+                inputMarkup, null,
+                callback: () =>
+                {
+                    notHandled = true;
+                });
+
+            Assert.True(notHandled);
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitStartOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //[||] Test Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        //
+        // Test Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitMiddleOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test [||]Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test 
+        //Comment
+    }
+}");
+        }
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitEndOfComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test Comment[||] 
```

Comment:
If my earlier point about trimming whitespace at the split is addressed, then I would expect this case to act as if the cursor were at the end of the line, so the trailing whitespace here would get dropped and you just get a new line with no comment.

Reply:
Hi @ArrowCase Thanks for your feedback. I totally agree with your suggestions and  I did not originally consider this use case.

I will look at implementing this soon, however I’m going to a bit short on time over the next week as I make these contributions in my free time, feel free to make a copy of this PR and implement it yourself if you want.